packages/grafana-ui/src/components/uPlot/MicroPlot.tsx

```diff
@@ -59,7 +67,7 @@ export class MicroPlot extends PureComponent<Props, State> {
       legend: {
         show: false,
       },
-      tzDate: ts => uPlot.tzDate(new Date(ts * 1000), 'Etc/UTC'),
+      tzDate: ts => uPlot.tzDate(new Date(ts * 1000), tz!.abbreviation),
```

Comment:
@ryantxu `uPlot.tzDate` is fairly expensive since it has to go through 2 date instantiations and parse, you may want to add a fast-path and simply return `new Date` if the specified timezone already matches the browser's timezone. i do have `Etc/UTC` special-cased, but not anything else.

more info: https://github.com/leeoniya/uPlot/issues/252#issuecomment-636649663

i've opened https://github.com/leeoniya/uPlot/issues/271 to maybe handle this lib-side.